### ISSUE TITLE
Huge memory leaks with vue components

### ISSUE THREAD

**Issue by GaraGentor** [GeekyAnts/vue-native-core#163]
What I have done to create the memory leak.

I have set up a fresh vue-native project with the vue-native cli. I created a v-for on a vue component like in the example from vue-native.io to display a todo list. My Project contains 2 buttons to fill and clear an array of objects. When I keep repeating the fill and clear process, then the memory is increasing but not releasing. The app will crash after repeating it many times due to a memory issue. In the memory heap all ReactVueComponents will be alive for ever. This seems to be a general issue with vue components.

So I created a git repository that you can easily recreate the issue. There is a readme to make the project running. I hope this is all you need.

Specs:

react: 16.6.3
react-native: 0.58.1
vue-native-core: 0.0.8
vue-native-helper: 0.0.11
vue-native-scripts: 0.0.16

https://github.com/GaraGentor/vue-native-memory-leak

This is the memory behavior with vue components

![Bildschirmfoto 2019-07-08 um 10 38 18](https://user-images.githubusercontent.com/39561325/60796886-92a0d880-a16e-11e9-9554-d14d995f4238.png)

and this with react components
![Bildschirmfoto 2019-07-08 um 10 40 17](https://user-images.githubusercontent.com/39561325/60796919-a2b8b800-a16e-11e9-9d2e-5e8e3e6db0df.png)

I found a work around that the memory got nearly cleared completly after clearing the list, but always remain a little bit of memory in use. This is the memory behavior of react-components, that import vue-components
![Bildschirmfoto 2019-07-12 um 10 20 41](https://user-images.githubusercontent.com/39561325/61113529-c7729f80-a48e-11e9-9d04-0321ca25cf63.png)
the memory will be cleared a little bit more.

The Parent component, that renders the vue children keep their children after clearing the list. After 10x filling and clearing the list the Parent contains 5000 children.
![Bildschirmfoto 2019-07-12 um 10 30 07](https://user-images.githubusercontent.com/39561325/61114274-4c11ed80-a490-11e9-959d-a49e8ab8b092.png)
The screenshots are made with a simlulator on a real device the behavior is still little bit different.

**Comment by vahidalvandi** [comment]
Oh yes

**Comment by RishabhKarnad** [comment]
@GaraGentor This is what I have found:
- The `$destroy` implementation in `core/instance/index.js` is a no-op. It seems the author of `react-vue` did this because they didn't want the method to do anything since the memory management was to be handled by React
- There is no code to remove child Vue instances from parent Vue instances. This would normally have been done by `$destroy`

As a fix, I've implemented a `$destroy` method to take care of the above. It is almost identical to the regular `$destroy` code with minor changes. Then I am calling `this.vm.$destroy` on the `ReactVueComponent`'s `componentWillUnmount`. I'll submit a PR soon.
